datascience/src/pipeline/parsers/flux/log_parsers.py

```diff
@@ -0,0 +1,195 @@
+
+    children = tagged_children(dis)
+
+    if "SpecifiedFACatch" in children:
+        catches = [parse_spe(spe) for spe in children["SpecifiedFACatch"]]
+        value["catches"] = catches
+
+    data = {"log_type": "DIS", "value": value}
+
+    return data
+
+
+def parse_coe(coe):
+    
+    children = tagged_children(coe)
+
+    value = {
+        "effortZoneEntryDatetimeUtc": get_text(coe,'.//ram:OccurrenceDateTime/udt:DateTime'),
+        "targetSpeciesOnEntry": get_text(coe,'.//ram:SpeciesTargetCode[@listID="TARGET_SPECIES_GROUP"]'),
+    }
+
+    if "RelatedFLUXLocation" in children:
+        value["faoZoneEntered"] = get_text(coe,'.//*[@schemeID="FAO_AREA"]')
+        value["economicZoneEntered"] = get_text(coe,'.//*[@schemeID="TERRITORY"]')
+        value["statisticalRectangleEntered"] = get_text(coe,'.//*[@schemeID="STAT_RECTANGLE"]')
+        value["effortZoneEntered"] = get_text(coe,'.//*[@schemeID="EFFORT_ZONE"]')
+
+    pos=get_element(coe,'.//ram:SpecifiedPhysicalFLUXGeographicalCoordinate')
+    if  pos != None:
+        value["latitude"] = try_float(get_text(pos,'.//ram:LatitudeMeasure'))
+        value["longitude"] = try_float(get_text(pos,'.//ram:LongitudeMeasure'))
+
+
+    data = {"log_type": "COE", "value": value}
+
+    return data
+
+
+def parse_cox(cox):
+    children = tagged_children(cox)
+
+    value = {
+        "effortZoneExitDatetimeUtc": get_text(cox,'.//ram:OccurrenceDateTime/udt:DateTime'),
+        "targetSpeciesOnExit": get_text(cox,'.//ram:SpeciesTargetCode[@listID="FAO_SPECIES"]'),
+    }
+
+    if "RelatedFLUXLocation" in children:
+        value["faoZoneExited"] = get_text(cox,'.//*[@schemeID="FAO_AREA"]')
+        value["economicZoneExited"] = get_text(cox,'.//*[@schemeID="TERRITORY"]')
+        value["statisticalRectangleExited"] = get_text(cox,'.//*[@schemeID="STAT_RECTANGLE"]')
+        value["effortZoneExited"] = get_text(cox,'.//*[@schemeID="EFFORT_ZONE"]')
+
+    pos=get_element(cox,'.//ram:SpecifiedPhysicalFLUXGeographicalCoordinate')
+    if pos != None:
+        value["latitudeExited"] = try_float(get_text(pos,'.//ram:LatitudeMeasure'))
+        value["longitudeExited"] = try_float(get_text(pos,'.//ram:LongitudeMeasure'))
+
+    data = {"log_type": "COX", "value": value}
+
+    return data
+
+
+def parse_pno(pno):
+
+    children = tagged_children(pno)
+
+    value = {
+        "predictedArrivalDatetimeUtc": get_text(pno,'.//ram:OccurrenceDateTime/udt:DateTime'),
+        "port": get_text(pno,'.//ram:RelatedFLUXLocation/ram:ID[@schemeID="LOCATION"]'),
+        "purpose": get_text(pno,'.//ram:ReasonCode'),
+        "tripStartDate": get_text(pno,'.//ram:EndDateTime/udt:DateTime'),
+    }
+    
+    if "RelatedFLUXLocation" in children:
+        for ras in children["RelatedFLUXLocation"]:
+            ras_data = parse_ras(ras)
+            value = {**value, **ras_data}
+
+    if "SpecifiedFACatch" in children:
+        catches = [parse_spe(spe) for spe in children["SpecifiedFACatch"]]
```

Comment:
Attention, toutes les captures dans un PNO de FLUX n'ont pas le même statut. Il y a d'une part les espèces à bord, et d'autre part les espèces qui vont être débarquées. Dans le champ `catches` du champ `value` du PNO, on ne veut que les espèces qui vont être débarquées.
Cf le messge de test Prior notification of arrival to port (Reason=Shelter).xml

Reply:
Corrigé en précisant `[ram:TypeCode='UNLOADED']` dans le sélecteur - la distinction entre espèces à bord et espèces à déarquer n'existe que dans FLUX et non dans l'ERS. Pour le moment on ne traite pas les données d'espèces à bord non destinées à être débarquées dans FLUX.